prog/weavewait/main.go

```diff
@@ -26,6 +26,11 @@ func main() {
 		args = args[1:]
 	}
 
+	if len(args) > 0 && args[0] == "-c" { // tell pid 1 to continue
+		checkErr(syscall.Kill(1, syscall.SIGUSR2))
```

Comment:
> Can you see a way to find the effective pid of a container, inside its namespace?

Nope.

I think it's worth going through Docker's container startup code to check whether the entrypoint could be executed as anything other than pid 1. Pretty sure the answer is 'no'.